RoboJackets/apiary-mobile
# Add basic BuzzCard reading functionality

- Adds a composable that can be used to add BuzzCard reading functionality to app screens
- Basic fleshed out version of the attendance UI, mostly non-functional at this point
- Added a top app bar
- Adds Apache 2.0 license for repository

**Testing instructions**

1. Open the app and sign in. You can sign into either the production or test environment.
   a. If you back out of the sign in window without completing it, a dialog should appear telling you that the authentication attempt was cancelled.
2. After signing in, you should be taken to the attendance screen.
3. Pressing back right now on the attendance screen should exit the app.
3. Reopen the app.  On the attendance screen, try tapping a BuzzCard.  "Processing..." should briefly appear, and then the total attendees counter should increment and the last attendee name should change to George P. Burdell.
4. If you tap something that isn't a BuzzCard, it should either cause the "We only support BuzzCards" error to appear or not recognize the NFC tag at all. Total attendees should not increment.
5. If you go to the Settings screen and return to attendance, the view state (last attendee, total attendees, etc.) should not reset.

**Screenshots**

![image](https://user-images.githubusercontent.com/5790137/127729510-585c2b76-51d0-498a-9dfe-189673c97370.png)
![image](https://user-images.githubusercontent.com/5790137/127729518-82e3edc2-2273-4a4a-b373-ad8c16e7e0f3.png)
![image](https://user-images.githubusercontent.com/5790137/127729522-f198265b-dc4e-49ef-913b-473ce39fc11c.png)


TODO:

- [x] Add error handling logic to NFC processing code
- [x] Add state within BuzzCardPrompt composable to move between normal and error
- [x] Wire BuzzCard prompt into attendance screen view model
- [x] ~Add NFC disabled detection~ scratched for this PR due to technical difficulties and put in #18
- [x] Manually create a release build for testing - Available in Slack
